src/app/dashboard/records/add-record-form.tsx

```diff
@@ -0,0 +1,271 @@
+/* eslint-disable @typescript-eslint/no-unsafe-call */
+/* eslint-disable @typescript-eslint/no-unsafe-assignment */
+"use client";
+
+import { zodResolver } from "@hookform/resolvers/zod";
+import { Check, ChevronsUpDown, Loader2, Plus } from "lucide-react";
+import { useState } from "react";
+import { useForm } from "react-hook-form";
+import * as z from "zod";
+
+import { Button } from "~/components/ui/button";
+import {
+  Command,
+  CommandEmpty,
+  CommandGroup,
+  CommandInput,
+  CommandItem,
+} from "~/components/ui/command";
+import {
+  Form,
+  FormControl,
+  FormDescription,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormMessage,
+} from "~/components/ui/form";
+import {
+  Popover,
+  PopoverContent,
+  PopoverTrigger,
+} from "~/components/ui/popover";
+import {
+  Select,
+  SelectContent,
+  SelectItem,
+  SelectTrigger,
+  SelectValue,
+} from "~/components/ui/select";
+import { toast } from "~/components/ui/use-toast";
+
+import { api } from "~/utils/api";
+import { cn } from "~/utils/shadcn";
+
+const FormSchema = z.object({
+  studentId: z.string(),
+  lectureId: z.string(),
+  attendanceStatus: z.union([
+    z.literal("PRESENT"),
+    z.literal("LATE"),
+    z.literal("ABSENT"),
```

Comment:
Shall we try using constants instead of hardcoded strings?

Reply:
Yeah I agree! So what we're doing here is using a validation library called `Zod` to create a constant literal value called `"PRESENT"`.

So it would _feel_ pretty redundant to use a constant literal value (using TypeScript) to create another constant literal value (using Zod)...but we'll be using `"PRESENT"` (and other statuses) throughout our codebase in frontend and backend so it would be useful to refactor it into a common constant that all files can use!

Reply:
Done!